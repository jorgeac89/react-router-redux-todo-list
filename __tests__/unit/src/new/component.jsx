import React from 'react';
import { shallow } from 'enzyme';
import New from 'new';

describe('New component', () => {
  const NewComponent = New.component;

  it('renders correctly without text', () => {
    expect(
      shallow(
        <NewComponent
          todoText=""
          onChangeTodoText={() => {}}
          onAddTodo={() => {}}
        />
      )
    ).toMatchSnapshot();
  });

  it('renders correctly with text', () => {
    expect(
      shallow(
        <NewComponent
          todoText="Example todo"
          onChangeTodoText={() => {}}
          onAddTodo={() => {}}
        />
      )
    ).toMatchSnapshot();
  });
});
