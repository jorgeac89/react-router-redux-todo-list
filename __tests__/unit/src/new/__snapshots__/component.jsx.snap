// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New component renders correctly with text 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex align-items-center justify-content-between"
    >
      <h2>
        New todo
      </h2>
      <Link
        className="font-weight-bold"
        to="/list"
      >
        <i
          className="fas fa-list-alt mr-2"
        />
        Todo list
      </Link>
    </div>
  </div>
  <form>
    <div
      className="row mt-3"
    >
      <div
        className="col-12"
      >
        <textarea
          className="form-control form-control-lg mb-2"
          onChange={[Function]}
          placeholder="Write the todo text here..."
          value="Example todo"
        />
      </div>
    </div>
    <div
      className="row mt-3"
    >
      <div
        className="col-12 text-center"
      >
        <button
          className="btn btn-primary font-weight-bold"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-save mr-2"
          />
          Save
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;

exports[`New component renders correctly without text 1`] = `
<Fragment>
  <div
    className="row"
  >
    <div
      className="col-12 d-flex align-items-center justify-content-between"
    >
      <h2>
        New todo
      </h2>
      <Link
        className="font-weight-bold"
        to="/list"
      >
        <i
          className="fas fa-list-alt mr-2"
        />
        Todo list
      </Link>
    </div>
  </div>
  <form>
    <div
      className="row mt-3"
    >
      <div
        className="col-12"
      >
        <textarea
          className="form-control form-control-lg mb-2"
          onChange={[Function]}
          placeholder="Write the todo text here..."
          value=""
        />
      </div>
    </div>
    <div
      className="row mt-3"
    >
      <div
        className="col-12 text-center"
      >
        <button
          className="btn btn-primary font-weight-bold"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-save mr-2"
          />
          Save
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;
